src/lib/coaler/embedder/SubstructureAnalyzer.cpp

```diff
@@ -0,0 +1,97 @@
+//
+// Created by chris on 12/7/23.
+//
+
+#include "SubstructureAnalyzer.hpp"
+
+#include <GraphMol/Substruct/SubstructMatch.h>
+#include <GraphMol/new_canon.h>
+
+namespace {
+    struct hasDegreeTwo {
```

Comment:
okay, aber ist das nötig, wenn es nur einmal genutzt wird?

Reply:
der struct operator wird auf allen atomen des mols in line 55 aufgerufen. Könnte natürlich auch ein lambda schreiben, aber so finde ich es deutlich leichter zu lesen.

Reply:
Fair point